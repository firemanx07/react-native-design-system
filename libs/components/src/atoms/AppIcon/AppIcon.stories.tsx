import { ComponentStory, ComponentMeta } from '@storybook/react';
import React from 'react';

import { AppIcon as AppIconBase, PropsType as AppIconProps } from './AppIcon';

const Story: ComponentMeta<typeof AppIconBase> = {
  component: AppIconBase,
  title: 'Atoms/AppIcon',
};
export default Story;

const Template: ComponentStory<typeof AppIconBase> = (args: AppIconProps) => (
  <AppIconBase {...args} />
);

const parameters = {
  controls: {
    include: ['size', 'isLight'],
  },
};

const defaultArgs: AppIconProps = {
  size: 85,
  isLight: false,
};

export const AppIcon = Template.bind({});
AppIcon.parameters = parameters;
AppIcon.args = {
  ...defaultArgs,
};
