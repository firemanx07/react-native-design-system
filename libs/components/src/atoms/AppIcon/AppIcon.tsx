import React, { memo } from 'react';

import AppIconLight from '../../assets/images/app-icon-light.svg';
import AppIconBase from '../../assets/images/app-icon.svg';
import { TestIDType } from '../../types';

export type PropsType = {
  size?: number;
  isLight?: boolean;
} & TestIDType;

export const AppIcon = ({ size = 85, isLight, testID }: PropsType) => {
  const appIconProps = {
    width: size,
    height: size,
    testID,
  };

  if (isLight) {
    return <AppIconLight {...appIconProps} />;
  }
  return <AppIconBase {...appIconProps} />;
};

export default memo(AppIcon);
