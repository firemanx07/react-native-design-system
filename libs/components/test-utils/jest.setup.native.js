import { Animated } from 'react-native';

require('react-native-reanimated/lib/reanimated2/jestUtils').setUpTests();

console.warn = () => {};

Animated.timing = () => ({
  start: () => jest.fn(),
});
