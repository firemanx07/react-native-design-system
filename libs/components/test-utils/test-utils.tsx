import { theme, ThemeProvider } from '@celebrate-app/themes';
import { Story } from '@storybook/react';
import {
  cleanup,
  render,
  RenderOptions,
  waitFor,
} from '@testing-library/react-native';
import React, { ReactNode } from 'react';

const Providers = ({ children }: { children: ReactNode }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

export const customRender = (
  ui: React.ReactElement<any>,
  options?: RenderOptions
) => {
  return render(ui, { wrapper: Providers, ...options });
};

beforeEach(cleanup);
afterEach(cleanup);

export function testStory(displayName: string, TestComponent: Story) {
  test(`${displayName} snapshot test`, async () => {
    const comp = customRender(<TestComponent {...TestComponent.args} />);
    jest.runOnlyPendingTimers();

    await waitFor(() => {
      expect(comp).toBeTruthy();
      expect(comp.toJSON()).toMatchSnapshot();
    });
  });
}

export function testComponentSnapshot(
  TestComponent: React.ReactElement<
    any,
    string | React.JSXElementConstructor<any>
  >
) {
  test('Snapshot test', () => {
    const comp = customRender(TestComponent);
    expect(comp).toBeTruthy();
    expect(comp.toJSON()).toMatchSnapshot();
  });
}

export function testAllStories(stories: any) {
  Object.keys(stories).forEach((story) => {
    if (story !== 'default') {
      testStory(story, stories[story]);
    }
  });
}
