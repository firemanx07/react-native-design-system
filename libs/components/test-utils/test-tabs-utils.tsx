import { waitFor } from '@testing-library/react-native';

import { customRender } from './test-utils';

type renderTabsProps = {
  elem: JSX.Element;
  testIDs: string[];
};
export const renderTabs = ({ elem, testIDs }: renderTabsProps) => {
  const comp = customRender(elem);
  const tabs: ReturnType<typeof comp.getByTestId>[] = [];
  testIDs.forEach(testID => tabs.push(comp.getByTestId(testID)));
  return { comp, tabs };
};
export const testTabs = async (args: ReturnType<typeof renderTabs>) => {
  await waitFor(() => {
    expect(args.comp).toBeTruthy();
    args.tabs.forEach(tab => expect(tab).toBeDefined());
  });
};
