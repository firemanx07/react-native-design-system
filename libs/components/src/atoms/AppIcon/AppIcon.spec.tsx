import { testAllStories } from '../../../test-utils';
import * as stories from './AppIcon.stories';

describe('AppIcon tests', () => {
  testAllStories(stories);
});
