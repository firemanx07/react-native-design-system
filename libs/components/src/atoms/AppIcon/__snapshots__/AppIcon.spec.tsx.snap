// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppIcon tests AppIcon snapshot test 1`] = `
<test-file-stub
  height={85}
  width={85}
/>
`;
